MoreLinq.Test/ShuffleTest.cs

```diff
@@ -0,0 +1,86 @@
+namespace MoreLinq.Test
+{
+    using System;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class ShuffleTest
+    {
+        static Random seed = new Random(12345);
+
+        [Test]
+        public void ShuffleIsLazy()
+        {
+            new BreakingSequence<int>().Shuffle();
+        }
+
+        [Test]
+        public void Shuffle()
+        {
+            var source = Enumerable.Range(1, 100);
+            var result = source.Shuffle();
+
+            Assert.That(result.OrderBy(x => x), Is.EqualTo(source));
+        }
+
+        [Test]
+        public void ShuffleWithEmptySequence()
+        {
+            var source = Enumerable.Empty<int>();
+            var result = source.Shuffle();
+
+            Assert.That(result, Is.Empty);
+        }
+
+        [Test]
+        public void ShuffleIsIdempotent()
```

Comment:
I'm not sure I quite understand what you are trying to test here and with `ShuffleSeedWithEmptySequence`. Could you explain? Is there a way the implementation could change that would cause this test to break?

Reply:
This test check that original source is not change (by side-effect) case It implements IList<T>. RandomSubset has the same test.

The test fails if we do the bellow

array = source as IList<TSource> ?? source.ToList().

Reply:
> `RandomSubset` has the same test.

I didn't see that and was looking at `Shuffle` on its own but thanks for clarifying. I now see from the comment in the same test for `RandomSubset`…

https://github.com/morelinq/MoreLINQ/blob/d49025012812436039805196a63663a258ded06d/MoreLinq.Test/RandomSubsetTest.cs#L186-L190

…what it's trying to test. I'm not sure if _idempotent_ is the right adjective here but that's a different story.